client/src/featureform/resources.py

```diff
@@ -1090,6 +1094,115 @@ def _create_feature_resource(self, db) -> None:
             self.variant,
             self.value_type
         )
+    
+    def _create_local_feature_type(self, db) -> None:
+        db.insert(
+            "feature_variant_category",
+            self.name,
+            self.variant,
+            "PRE_CALCULATED",
+        )
+
+    def get_status(self):
+        return ResourceStatus(self.status)
+
+    def is_ready(self):
+        return self.status == ResourceStatus.READY.value
+
+    def __eq__(self, other):
+        for attribute in vars(self):
+            if getattr(self, attribute) != getattr(other, attribute):
+                return False
+        return True
+
+
+class OnDemandFeatureDecorator:
+    def __init__(self,
+                 owner: str,
+                 tags: List[str] = None,
+                 properties: dict = {},
+                 variant: str = "default",
+                 name: str = "",
+                 description: str = "",
+                 status: str = "NO_STATUS"):
+        self.name = name
+        self.variant = variant
+        self.owner = owner
+        self.description = description
+        self.tags = tags
+        self.properties = properties
+        self.status = status
+
+    def __call__(self, fn):
+        if self.description == "" and fn.__doc__ is not None:
+            self.description = fn.__doc__
+        if self.name == "":
+            self.name = fn.__name__
+
+        self.query = dill.dumps(fn.__code__)
+        fn.name_variant = self.name_variant
+        fn.query = self.query
+        return fn
```

Comment:
@ahmadnazeri, one thing that I noticed with this that caught me off guard is that if you don't explicitly call the decorator, the query will be undefined at the time of creation. I suggest we capture it in `__init__` and then process any additional values that come if the user decides to call the decorator with args. Thoughts?

Reply:
can you please expand by what you mean when you say "capture it in `__init__`"? We can initialize it to None but wouldn't have access to fn until it's called.